Documentation/ABI/JSON.md

```diff
@@ -193,8 +193,8 @@ sufficient information to display the event in a human-readable format.
 }
 
 <event-kind> ::= "runStarted" | "testStarted" | "testCaseStarted" |
-  "issueRecorded" | "testCaseEnded" | "testEnded" | "testSkipped" |
-  "runEnded" ; additional event kinds may be added in the future
+  "issueRecorded" | "valueAttached" | "testCaseEnded" | "testEnded" |
```

Comment:
If `attachment.write` is used we need to capture the full path to the written file and include that in the JSON event payload. Currently there isn't a way for clients to know where an attachment was written. Example payload as it stands today:

```
{
    "kind": "event",
    "payload": {
        "_attachment": {},
        "instant": {
            "absolute": 763981.275023708,
            "since1970": 1729605021.633958
        },
        "kind": "valueAttached",
        "messages": [
            {
                "symbol": "attachment",
                "text": "Attached 'loremipsum.html' to \"With Attachment\"."
            }
        ],
        "testID": "EmptyAppTests.name()/EmptyAppTests.swift:18:2"
    },
    "version": 0
}
```

Reply:
We actually do capture the path as `"_attachment" -> "path"`, but you need to specify the directory to write attachments to, otherwise they don't actually get written out. That hasn't been plumbed through to SwiftPM yet, but you could do it by passing in a configuration JSON blob.

Reply:
Passing an absolute path to `attachment.write` seems to write the file out as I'd expect, but that path doesn't end up in `_attachment`.

Reply:
That's SPI for a Swift-based tool to use to write an attachment—as discussed off-GitHub, that's not the API surface you'll need to use here. :)